src/plugins/expressions/common/expression_types/specs/datatable.ts

```diff
@@ -42,19 +49,31 @@ export type DatatableColumnType = 'string' | 'number' | 'boolean' | 'date' | 'nu
  */
 export type DatatableRow = Record<string, any>;
 
+export interface DatatableColumnMeta {
+  type: DatatableColumnType;
+  field?: string;
+  params?: SerializableState | {};
+}
 /**
  * This type represents the shape of a column in a `Datatable`.
  */
 export interface DatatableColumn {
+  id: string;
   name: string;
-  type: DatatableColumnType;
+  meta: DatatableColumnMeta;
+}
+
+export interface DatatableMeta {
+  type?: string;
```

Comment:
Should `type` be required, not optional? That way even if we can't provide a source at least you have `type` to get a better idea as to where this table came from.

Reply:
i think we want it optional for all the cases where you get a datatable from unknown source